Automattic/wp-calypso
# Hosting: Allow user to switch to 'Beta' version in staging

See https://github.com/Automattic/dotcom-forge/issues/3248

## Proposed Changes

Adds some UI for a user to switch to the latest WordPress beta in their staging environment:

<img width="747" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/8b120b43-c72f-4a7e-a5ad-b617d7cfa7e8">

The production environment simply shows a message:

<img width="747" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/4685d07e-0e7d-4feb-a9ed-0cec86fbe711">


## Testing Instructions

1. Apply D117539-code to your sandbox.
2. Create a new Business site and take it Atomic.
3. Navigate to Hosting Configuration.
4. Verify the UI appears as expected.
5. Click 'Add staging site'.
6. Once the staging site is created, navigate to its Hosting Configuration page.
7. Verify you can now switch WordPress versions.
8. Switch to 'Beta'.
9. Verify your WordPress site is running the latest beta version.
